### PR TITLE
Project resource and datasources

### LINKED ISSUE #65: Project Support
Enterprise customer that would like to see project support and potentially even role support (as RBAC becomes more fine grain) in the Terraform provider.  This would give a powerful ability to properly manage (and template) projects properly, which currently can grow out of control without proper management/documentation because of a lack of visibility into each project consists of.

A simple workflow could be to create a project via a terraform provider with the for a development 'project'.  With support for the project to go into production that module is then used to create a production project.  Terraforms IAC module gives visability into what are in each project without messing with the Nutanix API to pull this information together.  It has become unwieldy to manage and remember what is in even as few as 25 project across a few clusters as my team as learned and this may help fix this big issue we have.

### PR DESCRIPTION
**Pull request by PacoDw**
Added:

- nutanix_project resource.
- nutanix_project resource acceptance testing.
- nutanix_project resource website documentation.
- nutanix_project singular data source.
- nutanix_project singular data source acceptance testing.
- nutanix_project singular data source website documentation.
- nutanix_projects plural data source.
- nutanix_projects plural data source acceptance testing.
- nutanix_projects plural data source website documentation.

## Example Usage

```hcl
resource "nutanix_subnet" "subnet" {
  cluster_uuid       = "<YOUR_CLUSTER_ID>"
  name               = "sunet_test_name"
  description        = "Description of my unit test VLAN"
  vlan_id            = 31
  subnet_type        = "VLAN"
  subnet_ip          = "10.250.140.0"
  default_gateway_ip = "10.250.140.1"
  prefix_length      = 24

  dhcp_options = {
    boot_file_name   = "bootfile"
    domain_name      = "nutanix"
    tftp_server_name = "10.250.140.200"
  }

  dhcp_domain_name_server_list = ["8.8.8.8", "4.2.2.2"]
  dhcp_domain_search_list      = ["terraform.nutanix.com", "terraform.unit.test.com"]
}

resource "nutanix_project" "project_test" {
  name        = "my-project"
  description = "This is my project"

  categories {
    name  = "Environment"
    value = "Staging"
  }

  resource_domain {
    resources {
      limit         = 4
      resource_type = "STORAGE"
    }
  }

  default_subnet_reference {
    uuid = nutanix_subnet.subnet.metadata.uuid
  }

  api_version = "3.1"
}

data "nutanix_project" "test" {
    project_id = nutanix_project.project_test.id
}

data "nutanix_projects" "test" {
    depends_on = ["nutanix_project.test"]
}
```

closes #65

### CODE REVIEW

**Comment by marinsalinas** [comment]
Could you provide a link to the API documentation to compare the Terraform schema vs the API schema?

**Comment by PacoDw** [comment]
@marinsalinas https://www.nutanix.dev/reference/prism_central/v3/api/projects/postprojects/

File: nutanix/data_source_nutanix_project_test.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+	subnetName := acctest.RandomWithPrefix("test-subnateName")
+	name := acctest.RandomWithPrefix("test-project-name-dou")
+	description := acctest.RandomWithPrefix("test-project-desc-dou")
+	categoryName := "Environment"
+	categoryVal := "Staging"
+	limit := cast.ToString(acctest.RandIntRange(2, 4))
+	rsType := "STORAGE"
+
+	updateName := acctest.RandomWithPrefix("test-project-name-dou")
```
Comment (marinsalinas) [inline nutanix/data_source_nutanix_project_test.go:23]: Nince!

**Review by marinsalinas — APPROVED** [review]
LGTM, Could you provide the screenshot of the ACCTest Passing?

**Comment by PacoDw** [comment]
@marinsalinas the test cases for the resource and datasources

projectpass

projectpass2

projectpass3

projectpass4

**Pull request merged**